agent/api/task/task.go

```diff
@@ -316,58 +316,16 @@ func TaskFromACS(acsTask *ecsacs.Task, envelope *ecsacs.PayloadMessage) (*Task,
 	//initialize resources map for task
 	task.ResourcesMapUnsafe = make(map[string][]taskresource.TaskResource)
 
-	// get the service connect attachment from acsTask.Attachments
-	if acsTask.Attachments != nil {
-		scConfig, err := getServiceConnectConfig(acsTask)
-		if err != nil {
-			return nil, err
-		}
-		task.ServiceConnectConfig = scConfig
+	// parse and validate value parsed from attachments
+	scConfig, err := HandleTaskAttachments(acsTask)
+	if err != nil {
+		return nil, err
 	}
+	task.ServiceConnectConfig = scConfig
```

Comment:
Could we move this as part of `HandleTaskAttachments` ?

Reply:
Move it to `HandleTaskAttachments`. Please see the update in commit 038dca8ba6b3da87074340d9ac535870a2ca39a6. Thanks.